src/include/duckdb/common/progress_bar/progress_bar.hpp

```diff
@@ -34,7 +34,10 @@ class ProgressBar {
 	void Update(bool final);
 	//! Gets current percentage
 	double GetCurrentPercentage();
-
+	//! Gets Current Rows Read
+	uint64_t GetCurrentRows();
```

Comment:
Perhaps instead of having two different functions here we can have a single function that returns a struct:

```cpp
struct ProgressData {
   double percentage;
   uint64_t rows_processed;
   uint64_t total_rows_to_process;
};

ProgressData GetDetailedProgressData();
```

That would be a cleaner/more extensible API in my opinion.

Reply:
Absolutely.